### ISSUE TITLE
[Bug] PeriodicKernel.lengthscale is not use as a lengthscale

### ISSUE THREAD

**Issue by vvvvalvalval** [cornellius-gp/gpytorch#1020]
# 🐛 Bug

**TL;DR:** [the implementation of `PeriodicKernel`](https://github.com/cornellius-gp/gpytorch/blob/f061240d087732d36b2a6f08a7caaefa07434fcd/gpytorch/kernels/periodic_kernel.py#L113) forgets to square its `lengthscale` parameter.

It seems to me that [the implementation of `PeriodicKernel`](https://github.com/cornellius-gp/gpytorch/blob/f061240d087732d36b2a6f08a7caaefa07434fcd/gpytorch/kernels/periodic_kernel.py#L113) forgets to square its `lengthscale` parameter, contrary to its [documentation](https://gpytorch.readthedocs.io/en/latest/kernels.html#periodickernel) and natural expectations.

**Resolutions strategies:** I suggest one of the following:

1. Change the implementation to square the `lengthscale` parameter. Keeps the API natural, but it's a breaking change. I don't think it's a big deal, since a significant bug was recently corrected, which suggests that the current kernel is not much relied upon.

2. Change the documentation to warn that the `lengthscale` is not squared, contrary to natural expectations. It's counter-intuitive, but avoids a breaking change.

3. Deprecate `PeriodicKernel`, and provide a new kernel e.g `PeriodicKernelV2`. Makes available a natural API without incurring breaking changes, but adds an ugly name. Could also be relevant because it's not that natural to view the parameter as a lengthscale anyway, as it has the dimension of an angle rather than a length in input space.

## To reproduce

** Code snippet to reproduce **

```python
## We demonstrate the bug by computing the covariance between 2 input points x1 and x2,
## using parameters and values so that the result is obviously wrong.

import torch
import gpytorch
import math

## BACKGROUND REFRESHER: semantically, the PeriodicKernel is like an RBF kernel
## applied to inputs that have been projected onto the unit circle
## using the appropriate period length.

## To make the bug obvious, we choose the lengthscale, period and inputs such that:
## (a) dist(c(x1), c(x2)) ≈ dist(x1, x2), where c(x) projects x to the unit circle.
## (b) dist(x1, x2) >> lengthscale, and therefore the correlation computed by the kernel should be close to 0.

l = 1e-2    ## the lengthscale
p = math.pi ## the period length

k = gpytorch.kernels.PeriodicKernel(active_dims=(0,)).initialize(lengthscale=l, period_length=p)

x1 = torch.Tensor([[0.]])
x2 = torch.Tensor([[5e-2]])

gpytorch_answer = k.forward(x1, x2).detach().item()
correct_answer = torch.exp((-2 / l**2) * torch.sin(math.pi * (x1 - x2) / p) ** 2).item()
```

** Stack trace/error message **

Not relevant here.

## Expected Behavior

Expected: numerical results consistent with the documentation.

## System information

**Please complete the following information:**
- GPyTorch Version (run `print(gpytorch.__version__)`: `1.0.0`
- PyTorch Version (run `print(torch.__version__)`: `1.3.1`
- Computer OS: macOS High Sierra 10.13.3

**Comment by gpleiss** [comment]
Yeah - this should be `lengthscale ** 2`. Since it is a breaking change we will need to add code that fixes models that might be loaded with the old parameterization.

> I don't think it's a big deal, since a significant bug was recently corrected, which suggests that the current kernel is not much relied upon.

Though no one corrected this parameterization, I don't think it's fair to assume that it is unused. In it's current state you can still learn decent periodic kernel hyperparameters with various models.

**Comment by vvvvalvalval** [comment]
Fair enough, if you want that level of commitment to compatibility I'd personally go for my strategy 3, but you are judge :)
